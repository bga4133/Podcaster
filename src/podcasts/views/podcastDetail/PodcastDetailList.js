import React from 'react'
import { Link } from 'react-router-dom'
import { secondsToTime } from '../../atoms/utils/secondsToTime'

export const PodcastDetailList = ({trackId , trackName, releaseDate, trackTimeMillis, getIdEpisode, podcastId}) => {
  return (
    <tr className='podcastDetail' key={trackId}>
    <td>
    <Link onClick={() => getIdEpisode(trackId)} className='titlePodcastList' to={`/podcast/${podcastId}/episode/${trackId}`}>Details
    {trackName}
    </Link>
    </td>
    <td>{releaseDate}</td>
    <td>{secondsToTime(trackTimeMillis)}</td>
  </tr>
  )
}
