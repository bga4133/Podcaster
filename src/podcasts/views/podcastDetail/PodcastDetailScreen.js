import React, { useEffect, useMemo, useState } from 'react'
import { Link, useParams} from 'react-router-dom';
import { Loader } from '../../../components/loader/Loader';
import { secondsToTime } from '../../atoms/utils/secondsToTime';
import { Sidebar } from '../components/Sidebar';
import { PodcastDetailList } from './PodcastDetailList';


export const PodcastDetailScreen = () => {
const [PodcastResponseDetail, setPodcastResponseDetail] = useState([]); 
const [IsLoading, setIsLoading] = useState(false);
  
const { podcastId } = useParams();

const getPodcastDetailById2 = async () => {
  setIsLoading(true);
  try {
      const response = await fetch (`https://cors-anywhere.herokuapp.com/itunes.apple.com/lookup?id=${podcastId}&media=podcast&entity=podcastEpisode&limit=100`);
      const data = await response.json();
      const { results } = data;
      results.slice(1);
      //results.slice(1), episodiUrl
      setPodcastResponseDetail(results)
      setIsLoading(false);
  } catch (e) {
      setIsLoading(false);
      throw new Error(e);
  }
}
useEffect(() => {
  getPodcastDetailById2();
}, [])

let podcastDetail = useMemo(() => PodcastResponseDetail, [PodcastResponseDetail]);
const PrincipalInfo = podcastDetail.filter((item) => item.wrapperType === 'track');
localStorage.setItem("PrincipalInfo", JSON.stringify(PrincipalInfo));
const descriptionInfo = podcastDetail.filter((item) => item.description)
localStorage.setItem("PrincipalInfoDescription", JSON.stringify(descriptionInfo[2]));

const getIdEpisode = (id) => {
  const episodeById = podcastDetail.filter((item) => id === item.trackId);
  localStorage.setItem("episodeById", JSON.stringify(episodeById));
}
  return (
    <div key="" className='containerPodcastDetail'>
      {IsLoading && <Loader />}
      <div className="columnInfoDetail">
      {PrincipalInfo.map(({colectionId, artworkUrl600, artistName, trackCount}) => (
        <Sidebar colectionId={colectionId} artworkUrl600={artworkUrl600} artistName={artistName} trackCount={trackCount} />
      ))}
      </div>
      <div className='columAudioInfo'>
        <div className='cardAudioInfo' data-testid="imgId">
          <div>
          {PrincipalInfo.map((item) => (
                <div key={item.trackId}>
                    <h2>Episodes: {item.trackCount}</h2>
                </div>
                ))}
          </div>
        <table className='table'>
          <thead>
            <tr>
              <th>Title</th>
              <th>Date</th>
              <th>Duration</th>
            </tr>
          </thead>
         <tbody>
          {podcastDetail.map(({trackId , trackName, releaseDate, trackTimeMillis}) => ( 
            <PodcastDetailList podcastId={podcastId} getIdEpisode={getIdEpisode} key={trackId} trackId={trackId} trackName={trackName} releaseDate={releaseDate} trackTimeMillis={trackTimeMillis} />
          ))}
          </tbody>
      </table>
        </div>
      </div>
    </div>
  )
}
