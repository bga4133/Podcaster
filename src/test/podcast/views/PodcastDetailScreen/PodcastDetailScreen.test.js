import { render, screen, waitFor } from "@testing-library/react"
import { PodcastDetailScreen } from "../../../../podcasts/views/podcastDetail/PodcastDetailScreen";

describe('<PodcastScreen /> test', () => {
    const { container } = render(<PodcastDetailScreen />);
    it("render the component", async () => {
        expect(container).toMatchSnapshot();
    })
    it("shouldn't show the div with the class podcastScreen if data is no loadet", async () => {
        const containerDiv = container.querySelector('podcastScreen');
        expect(containerDiv).toBeFalsy();
    });
    it("should show the data after load the page ", async () => {
        const { container } = render(<PodcastDetailScreen  />);
        const containerDiv = container.querySelector("container");
        await waitFor(
          () => {
            expect(screen.getByTestId("imgId")).toBeInTheDocument();
          },
          { timeout: 4000 }
        );
      });
    
});