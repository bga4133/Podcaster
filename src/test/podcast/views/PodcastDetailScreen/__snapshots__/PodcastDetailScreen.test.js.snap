// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PodcastScreen /> test render the component 1`] = `
<div>
  <div
    class="containerPodcastDetail"
  >
    <div
      class="lds-dual-ring"
    />
    <div
      class="columnInfoDetail"
    />
    <div
      class="columAudioInfo"
    >
      <div
        class="cardAudioInfo"
        data-testid="imgId"
      >
        <div />
        <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                Title
              </th>
              <th>
                Date
              </th>
              <th>
                Duration
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
